pallets/parachain-staking/src/lib.rs

```diff
@@ -2865,6 +2873,27 @@ pub mod pallet {
 				log::error!("💥 keeping old session because of empty collator set!");
 				None
 			} else {
+				// take snapshot of these collators' staking info
+				for collator in collators.iter() {
+					let collator_state = CandidatePool::<T>::get(collator).unwrap();
+					<AtStake<T>>::insert(new_index, collator, collator_state);
+				}
+
+				let old_index = new_index - 1;
+				// [TODO] what to do with this returned weight?
+				let (_, total_stake) = Self::get_total_collator_staking_num(old_index);
+				let pot = Self::account_id();
+				let total_issuance = T::Currency::free_balance(&pot)
+					.checked_sub(&T::Currency::minimum_balance())
+					.unwrap_or_else(Zero::zero);
+
+				// take snapshot of previous session's staking totals for payout calculation
+				DelayedPayoutInfo::<T>::put(DelayedPayoutInfoT {
+					round: old_index,
+					total_stake,
+					total_issuance,
```

Comment:
How about the transaction fee?

Reply:
Good point, one i've overlooked looks like it. I'll look into that.

Reply:
After discussion, we already deal with that